pkg/api/defaults.go

```diff
@@ -118,8 +120,17 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpdate bool) {
 		if o.KubernetesConfig.KubernetesImageBase == "" {
 			o.KubernetesConfig.KubernetesImageBase = cloudSpecConfig.KubernetesSpecConfig.KubernetesImageBase
 		}
-		if o.KubernetesConfig.EtcdVersion == "" || isUpdate {
+		if o.KubernetesConfig.EtcdVersion == "" {
```

Comment:
@CecileRobertMichon What do you think of this implementation for etcd? Basically, we allow folks to reliably run newer versions of etcd if they wish. I think this makes sense for now, for etcd, as we are explicitly defaulting to a "not newest" version; there are good reasons why someone may want to override our default policy for their control plane.

For moby and containerd below, as of now we basically release the newest version (or at least, we want to), so we probably don't want to be as tolerant there.

Thoughts?

I'll update the docs accordingly after we determine what our strategy is here for these components.

Reply:
we should unit test this logic though

Reply:
unit tests are there, need to update docs